crates/ra_hir/src/code_model.rs

```diff
@@ -339,10 +339,14 @@ pub struct Struct {
 }
 
 impl Struct {
-    pub fn module(self, db: &impl HirDatabase) -> Module {
+    pub fn module(self, db: &impl DefDatabase) -> Module {
         self.id.module(db)
     }
 
+    pub fn krate(self, db: &impl DefDatabase) -> Option<Crate> {
+        self.module(db).krate(db)
+    }
```

Comment:
Heh, we really should add some traits around these

Reply:
OTOH, I am wondering if the absolutely huge hir compile times are due to our use of traints and Froms?

Reply:
See https://github.com/EmbarkStudios/rust-ecosystem/issues/42#issuecomment-533822802

Reply:
Yeah, I was thinking about doing that as well.